### ISSUE TITLE
Build under Windows

### ISSUE THREAD

**Issue by henorek** [meganz/android#2]
Hi there, could you be so kind and add example for Makefile configuration under Windows platform? I have problem with setting NDK_ROOT. To be more specific, I got this error even after setting NDK_ROOT to my NDK location on disk `NDK_ROOT=D:/ProgramFiles/Android/sdk/ndk-build`

```
Error:Execution failed for task ':app:cleanMegaSDK'.
> A problem occurred starting process 'command 'make''
```

**Comment by jraragon** [comment]
Hi, 

here you have an Android example: https://github.com/meganz/sdk/tree/master/examples/android/ExampleApp_AS

If you cannot build the SDK, you can always use the prebuilt libraries that you will find on that link

Thanks!

**Comment by henorek** [comment]
Yes I used it, but even after pasting prebuild library into project as instruction says it doesn't work, still got ```
Error:Execution failed for task ':app:cleanMegaSDK'.
> A problem occurred starting process 'command 'make''
```
error

**Comment by jraragon** [comment]
Ah, yes.. you are right. 

If you are going to use the prebuilt library, then you will have to comment the compilation in the build.gradle: https://github.com/meganz/sdk/blob/master/examples/android/ExampleApp_AS/app/build.gradle

These lines are the ones related to the compilation of the library:
```
task megaSDK(type: Exec, description: 'Compile MEGA SDK via NDK') {
        workingDir 'src/main/jni'
        commandLine 'make'
    }

    tasks.withType(JavaCompile) {
        compileTask -> compileTask.dependsOn megaSDK
    }

    task cleanMegaSDK(type: Exec, description: 'Clean MEGA SDK object files') {
        workingDir 'src/main/jni'
        commandLine 'make', 'veryclean'
    }

    clean.dependsOn 'cleanMegaSDK'
```
If you comment the `compileTask -> compileTask.dependsOn megaSDK` it will be enough, but if you want to comment all that piece of code it will be OK too.

**Comment by henorek** [comment]
Ok then, we are getting closer. Now it doesn't see Mega dependencies, like MegaApiJava etc, of course I pasted libraries into project. I'm trying to build Android client btw.

**Comment by jraragon** [comment]
The java files are on the SDK. Path: bindings/java/nz/mega/sdk

You have to include them on your project.

**Comment by henorek** [comment]
Got it. Still some problems, it looks like it cannot resolve some methods from MegaApi

```
E:\Projekty\AndroidStudioProjects\android\app\src\main\java\nz\mega\sdk\MegaApiJava.java
Error:(1022, 20) error: cannot find symbol method getMyUser()
Error:(3261, 16) error: cannot find symbol method enableTransferResumption()
Error:(3297, 16) error: cannot find symbol method enableTransferResumption(String)
Error:(3308, 16) error: cannot find symbol method disableTransferResumption()
Error:(3321, 16) error: cannot find symbol method disableTransferResumption(String)
Error:(4454, 39) error: no suitable method found for search(String)
method MegaApi.search(MegaNode,String,boolean) is not applicable
(actual and formal argument lists differ in length)
method MegaApi.search(MegaNode,String) is not applicable
(actual and formal argument lists differ in length)
Error:(4525, 23) error: cannot find symbol method authorizeNode(MegaNode)
E:\Projekty\AndroidStudioProjects\android\app\src\main\java\mega\privacy\android\app\lollipop\ManagerActivityLollipop.java
Error:(11500, 33) error: cannot find symbol variable CHANGE_TYPE_EMAIL
```

**Comment by CodeNextTechnologies** [comment]
I am also getting the same issue by trying both methods described in README file.

Any solution ?

**Comment by johnx32** [comment]
Thank you for commenting, for some time I have tried to compile this project but every time I catch a stone in the way, the last problem just in the last line of the build.sh XD
<pre>
$ {NDK_BUILD} -j8
</ pre>
I get this

<pre>
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega/posix/megasys.h:45:
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:824:9: error: use of undeclared identifier 'fseeko64'
    if (fseeko64(__file_, __width > 0 ? __width * __off : 0, __whence))
        ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:197:5: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::seekoff' requested here
    basic_filebuf();
    ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:1019:14: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::basic_filebuf' requested here
    explicit basic_ifstream(const char* __s, ios_base::openmode __mode = ios_base::in);
             ^
/home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/src/posix/fs.cpp:1425:14: note: in instantiation of member function
      'std::__ndk1::basic_ifstream<char, std::__ndk1::char_traits<char> >::basic_ifstream' requested here
    ifstream infile(configFile);
             ^
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/src/posix/fs.cpp:26:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega.h:33:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega/types.h:45:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega/posix/megasys.h:45:
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:826:20: error: use of undeclared identifier 'ftello'
    pos_type __r = ftello(__file_);
                   ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:842:9: error: use of undeclared identifier 'fseeko64'
    if (fseeko64(__file_, __sp, SEEK_SET))
        ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:197:5: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::seekpos' requested here
    basic_filebuf();
    ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:1019:14: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::basic_filebuf' requested here
    explicit basic_ifstream(const char* __s, ios_base::openmode __mode = ios_base::in);
             ^
/home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/src/posix/fs.cpp:1425:14: note: in instantiation of member function
      'std::__ndk1::basic_ifstream<char, std::__ndk1::char_traits<char> >::basic_ifstream' requested here
    ifstream infile(configFile);
             ^
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/src/posix/fs.cpp:26:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega.h:33:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega/types.h:45:
In file included from /home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/include/mega/posix/megasys.h:45:
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:906:13: error: use of undeclared identifier 'fseeko64'
        if (fseeko64(__file_, -__c, SEEK_CUR))
            ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:197:5: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::sync' requested here
    basic_filebuf();
    ^
/home/john/Android/Ndk/sources/cxx-stl/llvm-libc++/include/fstream:1019:14: note: in instantiation of member function
      'std::__ndk1::basic_filebuf<char, std::__ndk1::char_traits<char> >::basic_filebuf' requested here
    explicit basic_ifstream(const char* __s, ios_base::openmode __mode = ios_base::in);
             ^
/home/john/AndroidStudioProjects/android/app/src/main/jni/mega/sdk/src/posix/fs.cpp:1425:14: note: in instantiation of member function
      'std::__ndk1::basic_ifstream<char, std::__ndk1::char_traits<char> >::basic_ifstream' requested here
    ifstream infile(configFile);
             ^
4 errors generated.
make: *** [/home/john/AndroidStudioProjects/android/app/src/main/obj/local/armeabi-v7a/objs/megasdk/sdk/src/posix/fs.o] Error 1
make: *** Waiting for unfinished jobs....

</pre>

**Comment by dknchris** [comment]
No support for building on Windows apparently. I'm always getting 
`bash: /d/Android/Sdk/ndk/20.0.5594570/ndk-which: No such file or directory` even though ndk-which.cmd is present.
